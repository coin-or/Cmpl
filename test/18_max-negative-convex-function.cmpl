# 
# Calculation of the maximum of a negative convex function 
# f(x) := - 12 - 12 * xn^3 - 12 * xn^5
#
parameters:
	#distance epsilon
	e:=0.02;	
	#initial solution
	xu:= 0;
	xo:= 2;
	xn:= (xu+xo)/2;

	{ (xo-xu) > e :
		fd:= 12 - 12 * xn^3 - 12 * xn^5;
		{ fd >= 0 : xu:=xn; |
		  fd <= 0 : xo:=xn ;}
		xn:= (xu+xo)/2;
		
		fx := 12 * xn -3 * xn^4 - 2 * xn^6;		 
		echo "f'(xn): " + format("%10.4f",fd) + " xu: " + format("%6.4f",xu) + 
			" xo: " + format("%6.4f",xo) + " xn: " + format("%6.4f",xn) + 
			" f(xn): " + format("%6.4f",fx);
		repeat;
	}
	echo "Optimal solution found";
		x:=xn;
	echo "x: "+ format("%2.3f",x); 
	echo "function value: " + (12 * x -3 * x^4 - 2 * x^6);
