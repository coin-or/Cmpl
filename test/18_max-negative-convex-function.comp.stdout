f'(xn):   -12.0000 xu: 0.0000 xo: 1.0000 xn: 0.5000 f(xn): 5.7812
f'(xn):    10.1250 xu: 0.5000 xo: 1.0000 xn: 0.7500 f(xn): 7.6948
f'(xn):     4.0898 xu: 0.7500 xo: 1.0000 xn: 0.8750 f(xn): 7.8439
f'(xn):    -2.1940 xu: 0.7500 xo: 0.8750 xn: 0.8125 f(xn): 7.8672
f'(xn):     1.3144 xu: 0.8125 xo: 0.8750 xn: 0.8438 f(xn): 7.8829
f'(xn):    -0.3397 xu: 0.8125 xo: 0.8438 xn: 0.8281 f(xn): 7.8815
f'(xn):     0.5113 xu: 0.8281 xo: 0.8438 xn: 0.8359 f(xn): 7.8839
Optimal solution found
x: 0.836
function value: 7.883868
